include/xtensor/xpad.hpp

```diff
@@ -81,82 +81,104 @@ namespace xt
 
         using size_type = typename std::decay_t<E>::size_type;
         using value_type = typename std::decay_t<E>::value_type;
+        using return_type = temporary_type_t<value_type,
+                                             typename std::decay_t<E>::shape_type,
+                                             std::decay_t<E>::static_layout>;
 
-        auto out = e;
+        // place the original array in the center
 
+        auto new_shape = e.shape();
+        xt::xstrided_slice_vector sv;
```

Comment:
I think the right size can be resered to avoid potential dynamic allocations:
`sv.reserve(e.shape().size());`